Subject: [PATCH] new concept click notify to detail
---
Index: app/src/main/java/vn/com/vpbank/client/android/newconcept/login/VPLoginActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/newconcept/login/VPLoginActivity.java b/app/src/main/java/vn/com/vpbank/client/android/newconcept/login/VPLoginActivity.java
--- a/app/src/main/java/vn/com/vpbank/client/android/newconcept/login/VPLoginActivity.java	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/newconcept/login/VPLoginActivity.java	(date 1726199877326)
@@ -59,6 +59,7 @@
 import vn.com.vpbank.client.android.broadcast.NotificationEvent;
 import vn.com.vpbank.client.android.broadcast.NotificationService;
 import vn.com.vpbank.client.android.databinding.VpActivityLoginBinding;
+import vn.com.vpbank.client.android.fcm.ArgumentText;
 import vn.com.vpbank.client.android.navigation.DefaultStartFragmentContext;
 import vn.com.vpbank.client.android.newconcept.base.VPBaseSupportActivity;
 import vn.com.vpbank.client.android.newconcept.deeplink.VPDeepLinkConstant;
@@ -68,6 +69,7 @@
 import vn.com.vpbank.client.android.newconcept.login.component.ISShareData;
 import vn.com.vpbank.client.android.newconcept.login.helper.ISNavHelper;
 import vn.com.vpbank.client.android.newconcept.login.viewmodel.VPLoginViewModel;
+import vn.com.vpbank.client.android.newconcept.notification.VPNotificationFragment;
 import vn.com.vpbank.client.android.newconcept.operation.ISResult;
 import vn.com.vpbank.client.android.screenhelpler.ScreenChangeHelper;
 import vn.com.vpbank.client.android.ui.HandleDynamicLink;
@@ -76,6 +78,7 @@
 import vn.com.vpbank.client.android.ui.login.FirebaseCall;
 import vn.com.vpbank.client.android.ui.login.ResultAppAction;
 import vn.com.vpbank.client.android.ui.main_home.search.SearchRemoteData;
+import vn.com.vpbank.client.android.ui.notification.NotificationController;
 import vn.com.vpbank.client.android.ui.notification.NotificationLocalData;
 import vn.com.vpbank.client.android.ui.operation.login.MainScreenIntegratedStageBuilderProvider;
 import vn.com.vpbank.client.android.ui.operation.utils.ActivityStateHandler;
@@ -115,6 +118,7 @@
     private IdleReceiver mIdleReceiver;
     private Language language;
     private NavGraph mLoginNavGraph = null;
+    private String notifyID = null;
     private final CancellableMessageDisplayer mMessageDisplay = new CancellableMessageDisplayer() {
         @Override
         public void cancel() {
@@ -322,6 +326,14 @@
         }));
         /*[13867-14274][NC-Regular Login] - start operation at regular login for actions defined from IS - By NamNH76 07/03/2024 End*/
         with(viewModel.getResultNotificationEvent(), res -> res.observe(this, this::onReceivedEvent));
+        observeSingle(viewModel.getGetLocalUserInfoLiveData(), localUserInfo -> with(getViewBinding(), binding -> {
+            String jwt = NotificationController.getJwtFirebase(this, localUserInfo.getId());
+            if (TextUtils.isEmpty(jwt)) {
+                jwt = VPNotificationFragment.PRELOGIN;
+            }
+
+            NotificationController.requestGetNotificationDetail(this, null, notificationService, binding.progress, notifyID, jwt);
+        }));
     }
 
     @Override
@@ -332,6 +344,8 @@
     }
 
     private void handleDeepLinkDataBeforeLogin(Intent mainIntent) {
+        if (mainIntent == null) return;
+
         // case share extension QR
         if (mainIntent.hasExtra(VPDeepLinkConstant.SHARE_QR_EXTENSION_KEY)) {
             String shareQRUri = mainIntent.getStringExtra(VPDeepLinkConstant.SHARE_QR_EXTENSION_KEY);
@@ -344,15 +358,26 @@
         // bỏ qua nếu không có deeplink data
         if (deepLinkData == null) return;
 
-        // nếu type deeplink là navigate
+        // riêng type deeplink là navigate notification group thì xử lý ở activity luôn
         if (deepLinkData instanceof VPDeepLinkData.VPNavigateDeepLink) {
             VPDeepLinkData.VPNavigateDeepLink navigateDeepLink = (VPDeepLinkData.VPNavigateDeepLink) deepLinkData;
+            notifyID = mainIntent.getStringExtra(ArgumentText.NOTIFY_ID);
+            mainIntent.removeExtra(ArgumentText.NOTIFY_ID);
 
             if (TextUtils.isEmpty(navigateDeepLink.getDeviceId())) {
                 // nếu không phải case trust device => navigate notification group
                 if (navigateDeepLink.isNavigationToNotificationGroup()) {
-                    getNavController().navigate(R.id.action_replace_to_fragment_notification);
+                    if (TextUtils.isEmpty(notifyID)) {
+                        getNavController().navigate(R.id.action_replace_to_fragment_notification);
+                    }
                 }
+
+                // nếu có notifyID thì request get notify detail
+                if (!TextUtils.isEmpty(notifyID)) {
+                    if (getViewModel() != null) {
+                        getViewModel().requestUserStorage();
+                    }
+                }
             } else {
                 // case trust device
                 LoginRedirectScreenController.setRedirectScreenTypeAndNavigateToLoginAction(RedirectScreenType.TRUSTED_DEVICE,
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/build.gradle b/app/build.gradle
--- a/app/build.gradle	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/build.gradle	(date 1726134563146)
@@ -225,8 +225,8 @@
     dataBinding {
         enabled = true
     }
-    dynamicFeatures = [ ":card_module"]
-    assetPacks = [":on_demand_my_qr"]
+//    dynamicFeatures = [ ":card_module"]
+//    assetPacks = [":on_demand_my_qr"]
 
     configurations {
         all*.exclude group: 'com.google.guava', module: 'listenablefuture'
Index: app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPHandleDynamicLink.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPHandleDynamicLink.kt b/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPHandleDynamicLink.kt
--- a/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPHandleDynamicLink.kt	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPHandleDynamicLink.kt	(date 1726195798606)
@@ -243,8 +243,9 @@
         is VPDeepLinkData.VPShortCutsDeepLink -> VPShortCutsDeepLinkWorker()
         is VPDeepLinkData.VPAdjustDeepLink -> VPAdjustDeepLinkWorker()
         is VPDeepLinkData.VPPartnerDeepLink -> VPPartnerDeepLinkWorker()
-        is VPDeepLinkData.VPFirebaseDeepLink -> VPFirebaseDeepLinkWorker() // ưu tiên check trước URL deeplink
+        is VPDeepLinkData.VPFirebaseDeepLink -> VPFirebaseDeepLinkWorker() // Note: ưu tiên return trước URL deeplink
         is VPDeepLinkData.VPURLDeepLink -> VPURLDeepLinkWorker()
+//        is VPDeepLinkData.VPNavigateDeepLink -> xxx // Note: type navigate notification thì code trước new concept sẽ xử lý trên activity, nên không cần worker.
         else -> null
     }
 
Index: app/src/main/java/vn/com/vpbank/client/android/ocbrequest/ocbrequest/FirebaseService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/ocbrequest/ocbrequest/FirebaseService.java b/app/src/main/java/vn/com/vpbank/client/android/ocbrequest/ocbrequest/FirebaseService.java
--- a/app/src/main/java/vn/com/vpbank/client/android/ocbrequest/ocbrequest/FirebaseService.java	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/ocbrequest/ocbrequest/FirebaseService.java	(date 1726194715854)
@@ -30,6 +30,7 @@
 import vn.com.vpbank.client.android.ocbrequest.requestmodal.transfer_gift.TransferGiftWishDetailResponse;
 import vn.com.vpbank.client.android.ocbrequest.requestmodal.transfer_gift.TransferGiftWishListResponse;
 import vn.com.vpbank.client.android.ocbrequest.requestmodal.what_is_new.WhatIsNew;
+import vn.com.vpbank.client.android.ui.notification.modal.NotificationDetailResponse;
 import vn.com.vpbank.client.android.ui.notification.modal.NotificationsResponse;
 
 public interface FirebaseService {
@@ -113,6 +114,9 @@
     @POST("device/auditLogDevice")
     Call<ConfigIconResponse> getLog(@Body RequestBody bodyBanner);
 
+    @POST("get/notificationById")
+    Call<NotificationDetailResponse> getNotificationById(@Body RequestBody bodyBanner);
+
     @POST("/get/myQR")
     Call<BackgroundQrObject> getMyQR();
 
Index: settings.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/settings.gradle b/settings.gradle
--- a/settings.gradle	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/settings.gradle	(date 1726134563153)
@@ -7,9 +7,9 @@
 include ':view_common'
 include ':base_apps'
 include ':smart_otp'
-include ':card_module'
+//include ':card_module'
 //include ':lucky_money'
-include ':on_demand_my_qr'
+//include ':on_demand_my_qr'
 
 //include ':game_neo'
 
Index: app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkWorkerData.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkWorkerData.kt b/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkWorkerData.kt
--- a/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkWorkerData.kt	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkWorkerData.kt	(date 1726195798616)
@@ -11,12 +11,12 @@
     var notificationService: NotificationService? = null
 }
 
-// input data cho case xử lý deeplink TRƯỚC khi login
+// input data cho worker case xử lý deeplink TRƯỚC khi login
 class VPDeepLinkWorkerBeforeLogin(activity: FragmentActivity, deepLinkData: VPDeepLinkData?) :
     VPDeepLinkWorkerData(activity, deepLinkData) {
 }
 
-// input data cho case xử lý deeplink SAU khi login
+// input data cho worker case xử lý deeplink SAU khi login
 class VPDeepLinkWorkerAfterLogin(activity: FragmentActivity, deepLinkData: VPDeepLinkData? = null) :
     VPDeepLinkWorkerData(activity, deepLinkData) {
     var userSession: UserSession? = null
Index: app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkData.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkData.kt b/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkData.kt
--- a/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkData.kt	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/VPDeepLinkData.kt	(date 1726195798612)
@@ -44,6 +44,7 @@
     }
 }
 
+// box chứa data input cho case xử lý deeplink URL
 data class VPURLAuthNHandleData(val activity: FragmentActivity) {
     var notificationService: NotificationService? = null
     var userSession: UserSession? = null
Index: app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationPromotionViewFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationPromotionViewFragment.kt b/app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationPromotionViewFragment.kt
--- a/app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationPromotionViewFragment.kt	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationPromotionViewFragment.kt	(date 1726212118587)
@@ -20,6 +20,7 @@
 import vn.com.vpbank.client.android.databinding.NotificationPromotionViewFragmentBinding
 import vn.com.vpbank.client.android.navigation.StartFragmentContext
 import vn.com.vpbank.client.android.ui.HandleDynamicLink
+import vn.com.vpbank.client.android.ui.VPMainFullActivity
 import vn.com.vpbank.client.android.ui.notification.modal.NotificationPromotionMto
 import vn.com.vpbank.client.android.ui.support.BackButtonListener
 import vn.com.vpbank.client.android.ui.support.DateUtils
@@ -286,6 +287,9 @@
     }
 
     override fun onBackPressed(): Boolean {
+        if (activity is VPMainFullActivity) {
+            activity?.finish()
+        }
         FragmentUtils.remove(this)
         return true
     }
Index: app/src/main/java/vn/com/vpbank/client/android/ui/notification/modal/NotificationDetailResponse.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/ui/notification/modal/NotificationDetailResponse.java b/app/src/main/java/vn/com/vpbank/client/android/ui/notification/modal/NotificationDetailResponse.java
new file mode 100644
--- /dev/null	(date 1726194715868)
+++ b/app/src/main/java/vn/com/vpbank/client/android/ui/notification/modal/NotificationDetailResponse.java	(date 1726194715868)
@@ -0,0 +1,30 @@
+package vn.com.vpbank.client.android.ui.notification.modal;
+
+import com.google.gson.annotations.SerializedName;
+
+import vn.com.vpbank.client.android.ocbrequest.requestmodal.SerializableClass;
+
+public class NotificationDetailResponse extends SerializableClass {
+
+    @SerializedName("status")
+    private Boolean status;
+
+    @SerializedName("data")
+    private NotificationPromotionMto notificationPromotionMto;
+
+    public Boolean getStatus() {
+        return status;
+    }
+
+    public void setStatus(Boolean status) {
+        this.status = status;
+    }
+
+    public NotificationPromotionMto getNotificationPromotionMto() {
+        return notificationPromotionMto;
+    }
+
+    public void setNotificationPromotionMto(NotificationPromotionMto notificationPromotionMto) {
+        this.notificationPromotionMto = notificationPromotionMto;
+    }
+}
\ No newline at end of file
Index: app/src/main/java/vn/com/vpbank/client/android/newconcept/login/viewmodel/VPLoginViewModel.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/newconcept/login/viewmodel/VPLoginViewModel.java b/app/src/main/java/vn/com/vpbank/client/android/newconcept/login/viewmodel/VPLoginViewModel.java
--- a/app/src/main/java/vn/com/vpbank/client/android/newconcept/login/viewmodel/VPLoginViewModel.java	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/newconcept/login/viewmodel/VPLoginViewModel.java	(date 1726199877360)
@@ -145,11 +145,15 @@
     private Boolean mIsSuggestPW = false;
     private final MutableLiveData<ISResult> mRequestLogin = new MutableLiveData<>();
     private final MutableLiveData<OperationMetaMto> mOptMetaResponse = new MutableLiveData<>();
+    private final MutableLiveData<LocalUserInfo> mGetLocalUserInfoLiveData = new MutableLiveData<>();
 
     public MutableLiveData<ISResult> getRequestLogin() {
         return mRequestLogin;
     }
 
+    public MutableLiveData<LocalUserInfo> getGetLocalUserInfoLiveData() {
+        return mGetLocalUserInfoLiveData;
+    }
 
     public MutableLiveData<OperationMetaMto> getOptMetaResponse() {
         return mOptMetaResponse;
@@ -292,6 +296,20 @@
         requestLastLogin(mISShareData.getResultLastLogin()::postValue);
     }
 
+    public void requestUserStorage() {
+        doAsyncWithLifecycle(() -> {
+            if (asyncUserStorage != null) {
+                asyncUserStorage.requestLast(SendListeners.weak(new AbstractStatelessSendListener<>() {
+
+                    @Override
+                    public void onSuccess(VPLoginViewModel reference, LocalUserInfo lastLogin) {
+                        mGetLocalUserInfoLiveData.postValue(lastLogin);
+                    }
+                }, this));
+            }
+        });
+    }
+
     public boolean isErrorPW(OperationMessageMto message) {
         return message != null && null != message.getParameterId() && message.getParameterId().equals(LOGIN_MSG);
     }
Index: app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationController.java b/app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationController.java
--- a/app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationController.java	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/ui/notification/NotificationController.java	(date 1726199877352)
@@ -6,14 +6,18 @@
 import android.content.Context;
 import android.content.Intent;
 import android.os.Build;
+import android.os.Bundle;
 import android.text.TextUtils;
 import android.util.Base64;
 
 import androidx.activity.result.ActivityResultLauncher;
+import androidx.annotation.NonNull;
 
+import com.qulix.android.support.ui.ProgressPanel;
 import com.vpbank.android.secure_storage.VPBStorage;
 import com.vpbank.log_app.LogApp;
 
+import org.json.JSONException;
 import org.json.JSONObject;
 
 import java.text.SimpleDateFormat;
@@ -31,9 +35,11 @@
 import vn.com.vpbank.client.android.ApplicationProperties;
 import vn.com.vpbank.client.android.R;
 import vn.com.vpbank.client.android.UserSession;
+import vn.com.vpbank.client.android.broadcast.NotificationEvent;
 import vn.com.vpbank.client.android.broadcast.NotificationService;
 import vn.com.vpbank.client.android.navigation.StartFragmentContext;
 import vn.com.vpbank.client.android.navigation_editor.NavigationActivity;
+import vn.com.vpbank.client.android.newconcept.login.VPLoginActivity;
 import vn.com.vpbank.client.android.ocbrequest.network.FirebaseServiceGenerator;
 import vn.com.vpbank.client.android.ocbrequest.ocbrequest.FirebaseService;
 import vn.com.vpbank.client.android.ui.VPAuthActivity;
@@ -45,6 +51,7 @@
 import vn.com.vpbank.client.android.ui.more.mgm.GetMemberCampaignFragment;
 import vn.com.vpbank.client.android.ui.notification.modal.NotificationBalanceChangeHashMto;
 import vn.com.vpbank.client.android.ui.notification.modal.NotificationBalanceChangeMto;
+import vn.com.vpbank.client.android.ui.notification.modal.NotificationDetailResponse;
 import vn.com.vpbank.client.android.ui.notification.modal.NotificationPromotionMto;
 import vn.com.vpbank.client.android.ui.notification.modal.NotificationsResponse;
 import vn.com.vpbank.client.android.ui.payments.PaymentCategoryFragment;
@@ -85,6 +92,13 @@
         SharedPreferenceUtils.getInstance(context).setValue(SharedPreferenceUtils.KEY_NOTIFICATION_COUNT, count);
         ShortcutBadger.applyCount(context, count);
     }
+
+    public static void decreaseNotifyCount(Context context) {
+        int count = getNotifyCount(context) - 1;
+        if (count < 0) count = 0;
+        SharedPreferenceUtils.getInstance(context).setValue(SharedPreferenceUtils.KEY_NOTIFICATION_COUNT, count);
+        ShortcutBadger.applyCount(context, count);
+    }
 
     public static boolean isActiveViewFirstNotify(Context context, String userId) {
         return SharedPreferenceUtils.getInstance(context).getBoolanValue(userId + "_ACTIVE_VIEW_FIRST_BALANCE_CHANGE", false);
@@ -283,9 +297,6 @@
         syncActionBalanceChange(context, notificationBalanceChangeMto, "delete", jwt, progress);
     }
 
-    //Zzz
-
-
     public static String decodeContent(Context context, String contentEn, String userId) throws Exception {
 
         byte[] result = Base64.decode(contentEn, Base64.DEFAULT);
@@ -300,7 +311,6 @@
         return AESService.decrypt(contentKeyStr, encodeByte, ivKey);
     }
 
-
     // 28Feb2024 - Epic14836 - cuongvv: bổ sung multi button trong màn hình chi tiết thông báo
     public static void handleClickNotifyPromotionInsideAuthen(Context context, StartFragmentContext startFragmentContext,
                                                               UserSession userSession,
@@ -517,5 +527,60 @@
         }
     }
 
+    // 2024Aug9 - epic17808 - cuongvv6: click notify điều hướng vào detail
+    public static void requestGetNotificationDetail(Context context, StartFragmentContext startFragmentContext, NotificationService notificationService, ProgressPanel progressPanel, String notifyID, String jwt) {
+        // create request data
+        JSONObject paramBody = new JSONObject();
+        try {
+            paramBody.put("data", jwt);
+            paramBody.put("id", notifyID);
+        } catch (JSONException e) {
+            LogApp.e(e);
+        }
+        RequestBody requestBody = RequestBody.create(okhttp3.MediaType.parse("application/json"), paramBody.toString());
+
+        // create service
+        String firebaseKey = new ApplicationProperties(context).getFirebaseKey();
+        FirebaseService firebaseService = FirebaseServiceGenerator.createService(FirebaseService.class, firebaseKey, context);
+
+        // request
+        Call<NotificationDetailResponse> call = firebaseService.getNotificationById(requestBody);
+        if (progressPanel != null) progressPanel.showProgress();
+        call.enqueue(new Callback<>() {
+            @Override
+            public void onResponse(@NonNull Call<NotificationDetailResponse> call, @NonNull Response<NotificationDetailResponse> response) {
+                if (progressPanel != null) progressPanel.hideProgress();
+                if (!response.isSuccessful()
+                        || response.body() == null
+                        || response.body().getNotificationPromotionMto() == null) {
+                    LogApp.e("Request requestGetNotificationDetail: NOT SUCCESS ............");
+                    return;
+                }
+
+                NotificationPromotionMto notificationPromotionMto = response.body().getNotificationPromotionMto();
+                if (context instanceof VPLoginActivity) {
+                    HashMap<String, Object> data = new HashMap<>();
+                    data.put("notificationPromotionMto", notificationPromotionMto);
+                    VPMainFullActivity.start(context, NotificationPromotionViewFragment.class.getName(), data);
+//                    FragmentUtils.addChild(((VPLoginActivity) context).getSupportFragmentManager(), R.id.fragmentContainer,
+//                            NotificationPromotionViewFragment.newInstance(notificationPromotionMto,
+//                                    true, null, context));
+                } else {
+                    FragmentUtils.addChildFull(context, NotificationPromotionViewFragment.newInstance(notificationPromotionMto,
+                            false, startFragmentContext, context));
+                }
+                LogApp.e("Request requestGetNotificationDetail: SUCCESS ............");
 
+                // decrease notify count on badger
+                NotificationController.decreaseNotifyCount(context);
+                notificationService.notify(NotificationEvent.NOTIFICATION_BADGE_CHANGE, new Bundle());
+            }
+
+            @Override
+            public void onFailure(@NonNull Call<NotificationDetailResponse> call, @NonNull Throwable t) {
+                if (progressPanel != null) progressPanel.hideProgress();
+                LogApp.e(t);
+            }
+        });
+    }
 }
Index: app/src/main/java/vn/com/vpbank/client/android/newconcept/notification/VPNotificationFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/newconcept/notification/VPNotificationFragment.java b/app/src/main/java/vn/com/vpbank/client/android/newconcept/notification/VPNotificationFragment.java
--- a/app/src/main/java/vn/com/vpbank/client/android/newconcept/notification/VPNotificationFragment.java	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/newconcept/notification/VPNotificationFragment.java	(date 1726199877344)
@@ -82,7 +82,7 @@
     public static final String NOTIFICATION = "NOTIFICATION";
     public static final String BALANCE_CHANGING = "BALANCE_CHANGING";
     public static final String NOTIFICATION_PROMOTION = "NOTIFICATION_PROMOTION";
-    private final String PRELOGIN = "PreLogin";
+    public static final String PRELOGIN = "PreLogin";
     private final int delay = 3000; //milliseconds
     private final long maxNotiStrangeDevice = 5 * 60 * 1000;
     boolean isNotificationGone;
@@ -263,12 +263,13 @@
 
         }
 
+        // epic17539-cuongvv: clear notify count khi show màn hình ds notify, HuongNL10 đã confirm
         if (getActivity() instanceof VPLoginActivity) {
-            String userId = lastLogin != null ? lastLogin.getId() : "";
-            boolean isActiveViewFirstNotify = NotificationController.isActiveViewFirstNotify(getContext(), userId);
-            if (isActiveViewFirstNotify) {
+//            String userId = lastLogin != null ? lastLogin.getId() : "";
+//            boolean isActiveViewFirstNotify = NotificationController.isActiveViewFirstNotify(getContext(), userId);
+//            if (isActiveViewFirstNotify) {
                 withViewModel(obj -> obj.clearNotificationCount(getContext()));
-            }
+//            }
         }
     }
 
@@ -663,9 +664,9 @@
                             jwt = PRELOGIN;
                         }
                         initView(userId, jwt);
-                        if (isActiveViewFirst) {
+//                        if (isActiveViewFirst) {
                             viewModel.clearNotificationCount(getContext());
-                        }
+//                        }
                     }
                 });
 
Index: local.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/local.properties b/local.properties
--- a/local.properties	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/local.properties	(date 1726134564426)
@@ -4,9 +4,9 @@
 # Location of the SDK. This is only used by Gradle.
 # For customization when using a Version Control System, please read the
 # header note.
-#Wed Sep 04 17:40:58 ICT 2024
+#Thu Sep 12 16:49:24 ICT 2024
 keyAlias=vpbank
 keyPassword=cYb@L9va
-sdk.dir=D\:\\Setup\\sdk
+sdk.dir=C\:\\AndroidSDK
 storeFile=../vpbank_key.jks
 storePassword=cYb@L9va
Index: app/src/main/java/vn/com/vpbank/client/android/ui/VPMainFullActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/ui/VPMainFullActivity.java b/app/src/main/java/vn/com/vpbank/client/android/ui/VPMainFullActivity.java
--- a/app/src/main/java/vn/com/vpbank/client/android/ui/VPMainFullActivity.java	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/ui/VPMainFullActivity.java	(date 1726199877336)
@@ -171,8 +171,10 @@
 import vn.com.vpbank.client.android.ui.nickname.NicknameStatusFragment;
 import vn.com.vpbank.client.android.ui.notification.ChangeBalanceSettingFragment;
 import vn.com.vpbank.client.android.ui.notification.ListNotifyConfirmFragment;
+import vn.com.vpbank.client.android.ui.notification.NotificationPromotionViewFragment;
 import vn.com.vpbank.client.android.ui.notification.balancefilter.NotificationDetailBalanceFragment;
 import vn.com.vpbank.client.android.ui.notification.balancefilter.NotificationStatementFragment;
+import vn.com.vpbank.client.android.ui.notification.modal.NotificationPromotionMto;
 import vn.com.vpbank.client.android.ui.oob.OOBConfirmFragment;
 import vn.com.vpbank.client.android.ui.oob.OOBVerifyFragment;
 import vn.com.vpbank.client.android.ui.operation.card.morecredit.RegMoreCreditCardInputFragment;
@@ -650,7 +652,15 @@
             if (NiceAccountLoadScreenFragment.class.getName().equals(fragmentStartName)) {
                 return NiceAccountLoadScreenFragment.newInstance();
             }
-
+            if (NotificationPromotionViewFragment.class.getName().equals(fragmentStartName)) {
+                NotificationPromotionMto notificationPromotionMto = null;
+                if (fragmentStartData != null) {
+                    if (fragmentStartData.get("notificationPromotionMto") instanceof NotificationPromotionMto) {
+                        notificationPromotionMto = (NotificationPromotionMto) fragmentStartData.get("notificationPromotionMto");
+                    }
+                }
+                return NotificationPromotionViewFragment.newInstance(notificationPromotionMto,true, null, this);
+            }
 //        if (NiceAccountSearchFragment.class.getName().equals(fragmentStartName)) {
 //            return NiceAccountSearchFragment.instance();
 //        }
Index: app/src/main/java/vn/com/vpbank/client/android/fcm/ArgumentText.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/fcm/ArgumentText.java b/app/src/main/java/vn/com/vpbank/client/android/fcm/ArgumentText.java
--- a/app/src/main/java/vn/com/vpbank/client/android/fcm/ArgumentText.java	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/fcm/ArgumentText.java	(date 1726193880766)
@@ -4,4 +4,5 @@
     public static final String TRUSTED_DEVICE = "TRUSTED_DEVICE";
     public static final String EWALLET_DEVICE = "OutOfBand";
     public static final String DEVICE_ID = "deviceId";
+    public static final String NOTIFY_ID = "NOTIFY_ID";
 }
Index: app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/worker/VPURLDeepLinkWorker.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/worker/VPURLDeepLinkWorker.kt b/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/worker/VPURLDeepLinkWorker.kt
--- a/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/worker/VPURLDeepLinkWorker.kt	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/newconcept/deeplink/worker/VPURLDeepLinkWorker.kt	(date 1726195798597)
@@ -75,6 +75,7 @@
         LogApp.e("handleLinkInsideAuthN")
         LogApp.e(urlFull)
 
+        // tạo input data để cho worker từng S xử lý
         var authNHandleData: VPURLAuthNHandleData? = null
         try {
             // parse url data from raw url
@@ -112,8 +113,8 @@
 
         // call worker to handle authN deeplink data
         var isResolve = false
-        for (i in 0 until workers.size) {
-            if (workers[i].invoke(authNHandleData)) {
+        for (sWorker in workers) {
+            if (sWorker.invoke(authNHandleData)) {
                 isResolve = true
                 break
             }
Index: app/src/main/java/vn/com/vpbank/client/android/fcm/MyFirebaseMessagingService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/vn/com/vpbank/client/android/fcm/MyFirebaseMessagingService.java b/app/src/main/java/vn/com/vpbank/client/android/fcm/MyFirebaseMessagingService.java
--- a/app/src/main/java/vn/com/vpbank/client/android/fcm/MyFirebaseMessagingService.java	(revision 146ad85276d04c082cc6d78acb257a641eccfdee)
+++ b/app/src/main/java/vn/com/vpbank/client/android/fcm/MyFirebaseMessagingService.java	(date 1726193958539)
@@ -74,12 +74,13 @@
         String type = map.containsKey("type") ? String.valueOf(map.get("type")) : "";
         String body = map.containsKey("body") ? String.valueOf(map.get("body")) : "Quý khách có thông báo mới";
         String channelType = map.containsKey("channelType") ? String.valueOf(map.get("channelType")) : "";
+        String notifyID = map.containsKey("id") ? String.valueOf(map.get("id")) : "";
         if ("BALANCE_CHANGING".equals(type) || "NOTIFICATION".equals(type) || "PROMOTION".equals(type) || "DEBT_REMIND".equals(type) || "SYSTEM_NOTI".equals(type) || ArgumentText.TRUSTED_DEVICE.equals(type)) {
             //nếu là noti biến động số dư, noti all, promotion
             NotificationController.increaseNotifyCount(this);
             notificationService.notify(NotificationEvent.NOTIFICATION_BADGE_CHANGE, new Bundle());
 
-            showNotification(this, map, title, body, type, channelType);
+            showNotification(this, map, title, body, type, channelType, notifyID);
 //            if ("BALANCE_CHANGING".equals(type))
 //                notificationService.notify(NotificationEvent.NOTIFICATION_DATA_CHANGE, new Bundle());
         } else {
@@ -135,7 +136,7 @@
         return false;
     }
 
-    private void showNotification(Context mContext, Map<String, String> data, String title, String body, String type, String channelType) {
+    private void showNotification(Context mContext, Map<String, String> data, String title, String body, String type, String channelType, final String notifyID) {
         NotificationManager mNotificationManager;
 
         NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(mContext.getApplicationContext(), "Channel_Firebase");
@@ -183,6 +184,12 @@
         } else {
             putDataNavigationGroup(ii, type);
         }
+
+        // add notify ID
+        if (!TextUtils.isEmpty(notifyID)) {
+            ii.putExtra(ArgumentText.NOTIFY_ID, notifyID);
+        }
+
         int flags = PendingIntent.FLAG_UPDATE_CURRENT;
 //        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
 //            flags |= PendingIntent.FLAG_IMMUTABLE;
